astropy/io/ascii/tests/test_write.py

```diff
@@ -672,28 +673,28 @@ def test_latex_units():
     latexdict = copy.deepcopy(ascii.latexdicts["AA"])
     latexdict["units"] = {"NUV exp.time": "s"}
     out = StringIO()
-    expected = """\
-\\begin{table}{cc}
-\\tablehead{\\colhead{date} & \\colhead{NUV exp.time}\\\\ \\colhead{ } & \\colhead{s}}
-\\startdata
-a & 1 \\\\
-b & 2
-\\enddata
-\\end{table}
-""".replace(
-        "\n", os.linesep
+    tablehead = Template(
+        r"\tablehead{\colhead{date} & \colhead{NUV exp.time}\\ \colhead{$u1} & \colhead{$u2}}"
     )
-
+    expected = [
+        r"\begin{table}{cc}",
+        tablehead.substitute(u1=" ", u2="s"),
+        r"\startdata",
+        r"a & 1 \\",
+        "b & 2",
+        r"\enddata",
+        r"\end{table}",
+        "",
+    ]
```

Comment:
What about keeping triple quotes but indenting the content then calling dedent() before comparison? This might make it easier to change the expected output if needed?

Reply:
This test is tricky because we _are_ changing the expected output: both to account for `os.linesep` and to change the units. Using triple quotes means using `str.replace()` to make those changes and the way Black formats that is a bit awkward. The changes I'm proposing also make it very clear that changing the units should only affect a single specific line in the output, whereas using `str.replace()` does not make that obvious.

Git blame says that the last time the expected output was updated was in 2016, so I don't think the convenience of updating it in the future is worth worrying about.

Reply:
Initially I found the (dedented) triple-quoted block easier to parse as LaTeX, too, but actually I think getting rid of the extra `\`s makes up in readability for having to abstract from the lines-as-a-list in the new version.